--- pub/example.h
+++ exp/example.h
@@ -1,24 +1,36 @@
 #pragma once
 
+#include "__zz_cib_internal/example-predef.h"
+
 class A
 {
 public:
+  A(A&& rhs);
+public:
+  A(const A&);
   A();
-  virtual int AnotherVirtFunc() { return 100; } // New virtual added before existing one.
-  virtual int VirtFunc() { return 1; }
-  int SomeFunc() { return 10; }
-
-  virtual ~A() {}
+  virtual int AnotherVirtFunc();
+  virtual int VirtFunc();
+  int SomeFunc();
+  virtual ~A();
 
 private:
-  int x {100};
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(A, A);
 };
 
-class B : public virtual A
+class B : public ::A
 {
 public:
+  B(B&& rhs);
+public:
+  B(const B&);
+  ~B();
   B();
-  int VirtFunc() override  { return 15; }
+  virtual int VirtFunc();
+  static B* Create();
 
-  static B* Create() { return new B; }
+private:
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(B, B);
 };
+
+#include "__zz_cib_internal/example-postdef.h"
