--- pub/example.h
+++ exp/example.h
@@ -1,48 +1,47 @@
 #pragma once
 
-//! Contains example definitions to explain cib's functioning
-namespace Example {
+#include "__zz_cib_internal/example-predef.h"
 
+//! Contains example definitions to explain cib's functioning
+namespace Example
+{
 template <typename _T>
 class Number
 {
 public:
   Number(_T x)
     : x_(x)
   {
   }
   _T GetValue() const
   {
     return x_;
   }
   void SetValue(_T x)
   {
     x_ = x;
   }
-  void Add(const Number<_T>& num)
+    void Add(Number<_T> const& num)
   {
     x_ += num.x_;
   }
-
 private:
   _T x_;
 };
-
 class A
 {
 public:
-  A()
-    : x_(5)
-  {
-  }
-  void        Set(const Number<int>& x);
-  Number<int> Get() const
-  {
-    return x_;
-  }
+    A(A&& rhs);
+  public:
+    A(A const & );
+    ~A();
+    A();
+    void Set(Number<int> const & x);
+    Number<int> Get() const;
 
 private:
-  Number<int> x_;
+    __ZZ_CIB_CLASS_INTERNAL_DEF(A, Example::A);
 };
+}
 
-} // namespace Example
+#include "__zz_cib_internal/example-impl.h"
