--- pub/example.h
+++ exp/example.h
@@ -1,20 +1,35 @@
 #pragma once
 
+#include "__zz_cib_internal/example-predef.h"
+
 //! Example to see what cib does for classes with virtual methods.
 class A
 {
 public:
+  A(A&& rhs);
+public:
+  A(A const & );
   A();
   //! This is to know what cib does with virtual functions.
-  virtual int VirtFunc() { return 5; }
-  virtual ~A() {}
-};
+  virtual int VirtFunc();
+  virtual ~A();
 
-class B : public A
+private:
+  __ZZ_CIB_CLASS_INTERNAL_DEF(A, A);
+};
+class B : public ::A
 {
 public:
+  B(B&& rhs);
+public:
+  B(B const & );
+  ~B();
   B();
-  int VirtFunc() override { return 15; }
+  virtual int VirtFunc();
+  static B* Create();
 
-  static B* Create() { return new B; }
+private:
+  __ZZ_CIB_CLASS_INTERNAL_DEF(B, B);
 };
+
+#include "__zz_cib_internal/example-impl.h"
