--- ../0010-simple-class/cib/__zz_cib_Example-ids.h
+++ cib/__zz_cib_Example-ids.h
@@ -1,14 +1,14 @@
 #pragma once
 
 namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class258 {
-  //#= FullClassName: ::Example
+  //#= FullClassName: ::A
   enum { __zz_cib_classId = 258 };
 }}}
 
-namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class258 { namespace __zz_cib_Class259 {
-  //#= FullClassName: ::Example::A
+namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class259 {
+  //#= FullClassName: ::B
   enum { __zz_cib_classId = 259 };
-}}}}
+}}}
 
 namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class256 {
   //#= FullClassName: ::__zz_cib_stl_helpers
@@ -25,17 +25,37 @@
   enum { __zz_cib_internalClassId = 1 };
 }}
 
-namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class258 { namespace __zz_cib_Class259 {
+namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class258 {
   enum __zz_cib_Methodid {
-    //#= A(const ::Example::A&);
+    //#= A(const ::A&);
     __zz_cib_Copy_0 = 0,
+    //#= A();
+    __zz_cib_New_1 = 1,
+    //#= int VirtFunc();
+    VirtFunc_2 = 2,
     //#= ~A();
+    __zz_cib_Delete_3 = 3,
+    __zz_cib_nextMethodId = 4
+  };
+}}}
+
+namespace __zz_cib_ { namespace __zz_cib_ids { namespace __zz_cib_Class259 {
+  enum __zz_cib_Methodid {
+    //#= B(const ::B&);
+    __zz_cib_Copy_0 = 0,
+    //#= ~B();
     __zz_cib_Delete_1 = 1,
-    //#= A();
+    //#= B();
     __zz_cib_New_2 = 2,
-    //#= int SomeFunc();
-    SomeFunc_3 = 3,
-    __zz_cib_nextMethodId = 4
+    //#= int VirtFunc();
+    VirtFunc_3 = 3,
+    //#= ::B* Create();
+    Create_4 = 4,
+    //#= __zz_cib_CastTo__zz_cib_Class258
+    __zz_cib_CastTo__zz_cib_Class258 = 5,
+    //#= __zz_cib_CastFrom__zz_cib_Class258
+    __zz_cib_CastFrom__zz_cib_Class258 = 6,
+    __zz_cib_nextMethodId = 7
   };
-}}}}
+}}}
 
