--- ../010-simple-class/cib/example.h.cpp
+++ cib/example.h.cpp
@@ -18,20 +18,24 @@
   return new __zz_cib_Delegatee();
 }
 static int __zz_cib_decl SomeFunc(__zz_cib_Delegatee* __zz_cib_obj) {
   return __zz_cib_obj->::Example::A::SomeFunc();
 }
+static int __zz_cib_decl AnotherFunction(__zz_cib_Delegatee* __zz_cib_obj) {
+  return __zz_cib_obj->::Example::A::AnotherFunction();
+}
 }
 }}}
 
 namespace __zz_cib_ { namespace Example { namespace A {
 const __zz_cib_MethodTable* __zz_cib_GetMethodTable() {
   static const __zz_cib_MTableEntry methodArray[] = {
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_copy),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_delete),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_new),
-    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::SomeFunc)
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::SomeFunc),
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::AnotherFunction)
   };
-  static const __zz_cib_MethodTable methodTable = { methodArray, 4 };
+  static const __zz_cib_MethodTable methodTable = { methodArray, 5 };
   return &methodTable;
 }
 }}}
