--- pub/example.h
+++ exp/example.h
@@ -1,23 +1,23 @@
 #pragma once
 
+#include "__zz_cib_internal/example-predef.h"
+
 namespace Example
 {
   class A
   {
   public:
+    A(A&& rhs);
+  public:
+    A(const A&);
+    ~A();
     A();
-
-    A& operator << (int x) {
-      m = x;
-      return *this;
-    }
-
-    A& operator >> (int& x) {
-      x = m;
-      return *this;
-    }
+    A& operator <<(int x);
+    A& operator >>(int& x);
 
 private:
-  int m;
+    __ZZ_CIB_PROXY_CLASS_INTERNALS(A, Example::A);
   };
 }
+
+#include "__zz_cib_internal/example-postdef.h"
