--- interface-classes/pub/example.h
+++ interface-classes-and-abi-stability/pub/example.h
@@ -1,36 +1,39 @@
 #pragma once
 
 class Interface2;
 class Interface3;
 
 class Interface1
 {
 public:
+  virtual int disruptVTable() { return 1; }
   virtual Interface2* p() = 0;
   virtual Interface3& r() = 0;
   virtual ~Interface1() {}
 };
 
 class Interface2
 {
 public:
+  virtual int disruptVTable() { return 2; }
   virtual int f() = 0;
   virtual ~Interface2() {}
 };
 
 class Interface3
 {
 public:
+  virtual int disruptVTable() { return 3; }
   virtual int g() = 0;
   virtual ~Interface3() {}
 };
 
 class A
 {
 public:
   A();
   int SetInterface(Interface1* pInterface1) const
   {
     return pInterface1->p()->f() + pInterface1->r().g();
   }
 };
