--- pub/example.h
+++ exp/example.h
@@ -1,22 +1,35 @@
 #pragma once
 
+#include "__zz_cib_internal/example-predef.h"
+
 class Interface
 {
 public:
+  Interface(Interface&& rhs);
+public:
   virtual int Func() = 0;
-  virtual ~Interface() {}
+  virtual ~Interface();
 private:
   virtual int PrivateVirtualFunc() = 0;
+protected:
+  Interface();
 
-  friend class A;
+private:
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(Interface, Interface);
 };
 
 class A
 {
 public:
+  A(A&& rhs);
+public:
+  A(const A&);
+  ~A();
   A();
-  int UseInterface(Interface* pInterface) const
-  {
-    return pInterface->Func() + pInterface->PrivateVirtualFunc();
-  }
+  int UseInterface(Interface* pInterface) const;
+
+private:
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(A, A);
 };
+
+#include "__zz_cib_internal/example-postdef.h"
