--- ../simple-class/exp/__zz_cib_internal/example-impl.h
+++ exp/__zz_cib_internal/example-impl.h
@@ -5,12 +5,79 @@
 #include "__zz_cib_internal/__zz_cib_Example-mtable-helper.h"
 #include "__zz_cib_internal/__zz_cib_Example-handle-helper.h"
 
-namespace __zz_cib_ { namespace Example { namespace A {
+namespace __zz_cib_ { namespace A {
 class __zz_cib_Helper : public __zz_cib_::__zz_cib_MethodTableHelper
-  , public __zz_cib_::__zz_cib_HandleHelper<::Example::A, __zz_cib_Helper> {
+  , public __zz_cib_::__zz_cib_HandleHelper<::A, __zz_cib_Helper> {
 private:
-  friend class ::Example::A;
-  friend class __zz_cib_::__zz_cib_HandleHelper<::Example::A, __zz_cib_Helper>;
+  friend class ::A;
+  friend class __zz_cib_::__zz_cib_HandleHelper<::A, __zz_cib_Helper>;
+
+  __zz_cib_Helper()
+    : __zz_cib_::__zz_cib_MethodTableHelper(
+      __zz_cib_Example_GetMethodTable(__zz_cib_classid))
+  {}
+  static __zz_cib_Helper& instance() {
+    static __zz_cib_Helper helper;
+    return helper;
+  }
+
+  static __zz_cib_HANDLE* __zz_cib_new_0() {
+    using __zz_cib_newProc = __zz_cib_HANDLE* (__zz_cib_decl *) ();
+    return instance().invoke<__zz_cib_newProc, __zz_cib_methodid::__zz_cib_new_0>(
+      );
+  }
+  static __zz_cib_HANDLE* __zz_cib_copy_1(__zz_cib_HANDLE const * __zz_cib_param0) {
+    using __zz_cib_copyProc = __zz_cib_HANDLE* (__zz_cib_decl *) (__zz_cib_HANDLE const * __zz_cib_param0);
+    return instance().invoke<__zz_cib_copyProc, __zz_cib_methodid::__zz_cib_copy_1>(
+      __zz_cib_param0);
+  }
+  static int VirtFunc_2(__zz_cib_HANDLE* __zz_cib_obj) {
+    using VirtFuncProc = int (__zz_cib_decl *) (__zz_cib_HANDLE*);
+    return instance().invoke<VirtFuncProc, __zz_cib_methodid::VirtFunc_2>(
+      __zz_cib_obj
+      );
+  }
+  static int SomeFunc_3(__zz_cib_HANDLE* __zz_cib_obj) {
+    using SomeFuncProc = int (__zz_cib_decl *) (__zz_cib_HANDLE*);
+    return instance().invoke<SomeFuncProc, __zz_cib_methodid::SomeFunc_3>(
+      __zz_cib_obj
+      );
+  }
+  static void __zz_cib_delete_4(__zz_cib_HANDLE* __zz_cib_obj) {
+    if (__zz_cib_obj) {
+      using __zz_cib_deleteProc = void (__zz_cib_decl *) (__zz_cib_HANDLE*);
+      return instance().invoke<__zz_cib_deleteProc, __zz_cib_methodid::__zz_cib_delete_4>(
+        __zz_cib_obj
+        );
+    }
+  }
+  static ::A* __zz_cib_create_proxy(__zz_cib_HANDLE* h) {
+    return new ::A(h);
+  }
+public:
+    static ::A __zz_cib_obj_from_handle(__zz_cib_HANDLE* h) {
+      return ::A(h);
+    }
+    static __zz_cib_HANDLE*& __zz_cib_get_handle(::A* __zz_cib_obj) {
+      return __zz_cib_obj->__zz_cib_h_;
+    }
+    static __zz_cib_HANDLE* const& __zz_cib_get_handle(const ::A* __zz_cib_obj) {
+      return __zz_cib_obj->__zz_cib_h_;
+    }
+    static __zz_cib_HANDLE* __zz_cib_release_handle(::A* __zz_cib_obj) {
+      auto h = __zz_cib_obj->__zz_cib_h_;
+      __zz_cib_obj->__zz_cib_h_ = nullptr;
+      return h;
+    }
+};
+}}
+
+namespace __zz_cib_ { namespace B {
+class __zz_cib_Helper : public __zz_cib_::__zz_cib_MethodTableHelper
+  , public __zz_cib_::__zz_cib_HandleHelper<::B, __zz_cib_Helper> {
+private:
+  friend class ::B;
+  friend class __zz_cib_::__zz_cib_HandleHelper<::B, __zz_cib_Helper>;
 
   __zz_cib_Helper()
     : __zz_cib_::__zz_cib_MethodTableHelper(
@@ -39,29 +106,34 @@
         );
     }
   }
-  static int SomeFunc_3(__zz_cib_HANDLE* __zz_cib_obj) {
-    using SomeFuncProc = int (__zz_cib_decl *) (__zz_cib_HANDLE*);
-    return instance().invoke<SomeFuncProc, __zz_cib_methodid::SomeFunc_3>(
+  static int VirtFunc_3(__zz_cib_HANDLE* __zz_cib_obj) {
+    using VirtFuncProc = int (__zz_cib_decl *) (__zz_cib_HANDLE*);
+    return instance().invoke<VirtFuncProc, __zz_cib_methodid::VirtFunc_3>(
       __zz_cib_obj
       );
   }
-  static ::Example::A* __zz_cib_create_proxy(__zz_cib_HANDLE* h) {
-    return new ::Example::A(h);
+  static __zz_cib_HANDLE* __zz_cib_cast_to___A_4(__zz_cib_HANDLE* __zz_cib_obj) {
+    using __zz_cib_cast_to___AProc = __zz_cib_HANDLE* (__zz_cib_decl *) (__zz_cib_HANDLE* h);
+    return instance().invoke<__zz_cib_cast_to___AProc, __zz_cib_methodid::__zz_cib_cast_to___A_4>(__zz_cib_obj);
+  }
+  static ::B* __zz_cib_create_proxy(__zz_cib_HANDLE* h) {
+    return new ::B(h);
   }
 public:
-    static ::Example::A __zz_cib_obj_from_handle(__zz_cib_HANDLE* h) {
-      return ::Example::A(h);
+    static ::B __zz_cib_obj_from_handle(__zz_cib_HANDLE* h) {
+      return ::B(h);
     }
-    static __zz_cib_HANDLE*& __zz_cib_get_handle(::Example::A* __zz_cib_obj) {
+    static __zz_cib_HANDLE*& __zz_cib_get_handle(::B* __zz_cib_obj) {
       return __zz_cib_obj->__zz_cib_h_;
     }
-    static __zz_cib_HANDLE* const& __zz_cib_get_handle(const ::Example::A* __zz_cib_obj) {
+    static __zz_cib_HANDLE* const& __zz_cib_get_handle(const ::B* __zz_cib_obj) {
       return __zz_cib_obj->__zz_cib_h_;
     }
-    static __zz_cib_HANDLE* __zz_cib_release_handle(::Example::A* __zz_cib_obj) {
+    static __zz_cib_HANDLE* __zz_cib_release_handle(::B* __zz_cib_obj) {
       auto h = __zz_cib_obj->__zz_cib_h_;
       __zz_cib_obj->__zz_cib_h_ = nullptr;
+      __zz_cib_::A::__zz_cib_Helper::__zz_cib_release_handle(__zz_cib_obj);
       return h;
     }
 };
-}}}
+}}
