--- 010-simple-class/src/example-client.cpp
+++ 018-simple-class-and-fwd-compatibility/src/example-client.cpp
@@ -1,9 +1,18 @@
 #include "example.h"
 
 #include <catch/catch.hpp>
 
+#include <exception>
+
 TEST_CASE("Method call")
 {
   Example::A a;
-  CHECK(a.SomeFunc() == 1);
+  if(a.SomeFunc() == 2)
+  {
+    CHECK(a.AnotherFunction() == 109);
+  }
+  else
+  {
+    CHECK_THROWS_AS(a.AnotherFunction(), std::bad_function_call);
+  }
 }
