--- pub/example.h
+++ exp/example.h
@@ -1,37 +1,36 @@
 #pragma once
-
 #include <functional>
 
+#include "__zz_cib_internal/example-predef.h"
+
 class C
 {
 public:
-  C(int x);
-  virtual ~C() = default;
-
+  C(C&& rhs);
 public:
-  virtual int value() const {
-    return m;
-  }
+  C(const C& );
+  C(int x);
+  virtual ~C();
+  virtual int value() const;
 
 private:
-  int m;
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(C, C);
 };
 
 class A
 {
 public:
+  A(A&& rhs);
+public:
+  A(const A& );
+  ~A();
   A();
+  int f(std::function<int()> callback) const;
+  int g(const std::function<int()>& callback) const;
+  int h(std::function<int(C)> callback) const;
   
-public:
-  int f(std::function<int()> callback) const {
-    return 2 * callback();
-  }
-
-  int g(const std::function<int()>& callback) const {
-    return 5 * callback();
-  }
-
-  int h(std::function<int(C)> callback) const {
-    return callback(C(9));
-  }
+private:
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(A, A);
 };
+
+#include "__zz_cib_internal/example-postdef.h"
