--- ../080-complex-interface-classes/cib/example.h.cpp
+++ cib/example.h.cpp
@@ -27,10 +27,15 @@
   Interface1(__zz_cib_PROXY* proxy, const __zz_cib_MethodTable* mtbl)
     : ::Interface1::Interface1()
     , __zz_cib_proxy(proxy)
     , __zz_cib_mtbl_helper(mtbl)
   {}
+  int disruptVTable() override {
+    using disruptVTableProc = int (__zz_cib_decl *) (__zz_cib_PROXY*);
+    return __zz_cib_get_mtable_helper().invoke<disruptVTableProc, __zz_cib_GenericProxy::__zz_cib_methodid::disruptVTable>(
+      __zz_cib_proxy);
+  }
   ::Interface2* p() override {
     using pProc = ::Interface2* (__zz_cib_decl *) (__zz_cib_PROXY*);
     return __zz_cib_get_mtable_helper().invoke<pProc, __zz_cib_GenericProxy::__zz_cib_methodid::p>(
       __zz_cib_proxy);
   }
@@ -52,10 +57,13 @@
 namespace __zz_cib_Delegator {
 using __zz_cib_Delegatee = __zz_cib_::Interface1::__zz_cib_GenericProxy::Interface1;
 static ::Interface1* __zz_cib_decl __zz_cib_new(__zz_cib_PROXY* proxy, const __zz_cib_MethodTable* mtbl) {
   return new __zz_cib_::Interface1::__zz_cib_GenericProxy::Interface1(proxy, mtbl);
 }
+static int __zz_cib_decl disruptVTable(__zz_cib_Delegatee* __zz_cib_obj) {
+  return __zz_cib_obj->::Interface1::disruptVTable();
+}
 static ::Interface2* __zz_cib_decl p(__zz_cib_Delegatee* __zz_cib_obj) {
   return __zz_cib_obj->p();
 }
 static ::Interface3* __zz_cib_decl r(__zz_cib_Delegatee* __zz_cib_obj) {
   return &__zz_cib_obj->r();
@@ -76,13 +84,14 @@
   static const __zz_cib_MTableEntry methodArray[] = {
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_new),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::p),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::r),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_delete),
-    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_release_proxy)
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_release_proxy),
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::disruptVTable)
   };
-  static const __zz_cib_MethodTable methodTable = { methodArray, 5 };
+  static const __zz_cib_MethodTable methodTable = { methodArray, 6 };
   return &methodTable;
 }
 }}
 namespace __zz_cib_ { namespace Interface2 {
 namespace __zz_cib_GenericProxy {
@@ -99,10 +108,15 @@
   Interface2(__zz_cib_PROXY* proxy, const __zz_cib_MethodTable* mtbl)
     : ::Interface2::Interface2()
     , __zz_cib_proxy(proxy)
     , __zz_cib_mtbl_helper(mtbl)
   {}
+  int disruptVTable() override {
+    using disruptVTableProc = int (__zz_cib_decl *) (__zz_cib_PROXY*);
+    return __zz_cib_get_mtable_helper().invoke<disruptVTableProc, __zz_cib_GenericProxy::__zz_cib_methodid::disruptVTable>(
+      __zz_cib_proxy);
+  }
   int f() override {
     using fProc = int (__zz_cib_decl *) (__zz_cib_PROXY*);
     return __zz_cib_get_mtable_helper().invoke<fProc, __zz_cib_GenericProxy::__zz_cib_methodid::f>(
       __zz_cib_proxy);
   }
@@ -119,10 +133,13 @@
 namespace __zz_cib_Delegator {
 using __zz_cib_Delegatee = __zz_cib_::Interface2::__zz_cib_GenericProxy::Interface2;
 static ::Interface2* __zz_cib_decl __zz_cib_new(__zz_cib_PROXY* proxy, const __zz_cib_MethodTable* mtbl) {
   return new __zz_cib_::Interface2::__zz_cib_GenericProxy::Interface2(proxy, mtbl);
 }
+static int __zz_cib_decl disruptVTable(__zz_cib_Delegatee* __zz_cib_obj) {
+  return __zz_cib_obj->::Interface2::disruptVTable();
+}
 static int __zz_cib_decl f(__zz_cib_Delegatee* __zz_cib_obj) {
   return __zz_cib_obj->f();
 }
 static void __zz_cib_decl __zz_cib_delete(__zz_cib_Delegatee* __zz_cib_obj) {
   delete __zz_cib_obj;
@@ -151,13 +168,14 @@
   static const __zz_cib_MTableEntry methodArray[] = {
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_new),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::f),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_delete),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_get_class_id),
-    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_release_proxy)
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_release_proxy),
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::disruptVTable)
   };
-  static const __zz_cib_MethodTable methodTable = { methodArray, 5 };
+  static const __zz_cib_MethodTable methodTable = { methodArray, 6 };
   return &methodTable;
 }
 }}
 namespace __zz_cib_ { namespace Interface3 {
 namespace __zz_cib_GenericProxy {
@@ -174,10 +192,15 @@
   Interface3(__zz_cib_PROXY* proxy, const __zz_cib_MethodTable* mtbl)
     : ::Interface3::Interface3()
     , __zz_cib_proxy(proxy)
     , __zz_cib_mtbl_helper(mtbl)
   {}
+  int disruptVTable() override {
+    using disruptVTableProc = int (__zz_cib_decl *) (__zz_cib_PROXY*);
+    return __zz_cib_get_mtable_helper().invoke<disruptVTableProc, __zz_cib_GenericProxy::__zz_cib_methodid::disruptVTable>(
+      __zz_cib_proxy);
+  }
   int g() override {
     using gProc = int (__zz_cib_decl *) (__zz_cib_PROXY*);
     return __zz_cib_get_mtable_helper().invoke<gProc, __zz_cib_GenericProxy::__zz_cib_methodid::g>(
       __zz_cib_proxy);
   }
@@ -194,10 +217,13 @@
 namespace __zz_cib_Delegator {
 using __zz_cib_Delegatee = __zz_cib_::Interface3::__zz_cib_GenericProxy::Interface3;
 static ::Interface3* __zz_cib_decl __zz_cib_new(__zz_cib_PROXY* proxy, const __zz_cib_MethodTable* mtbl) {
   return new __zz_cib_::Interface3::__zz_cib_GenericProxy::Interface3(proxy, mtbl);
 }
+static int __zz_cib_decl disruptVTable(__zz_cib_Delegatee* __zz_cib_obj) {
+  return __zz_cib_obj->::Interface3::disruptVTable();
+}
 static int __zz_cib_decl g(__zz_cib_Delegatee* __zz_cib_obj) {
   return __zz_cib_obj->g();
 }
 static void __zz_cib_decl __zz_cib_delete(__zz_cib_Delegatee* __zz_cib_obj) {
   delete __zz_cib_obj;
@@ -226,13 +252,14 @@
   static const __zz_cib_MTableEntry methodArray[] = {
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_new),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::g),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_delete),
     reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_get_class_id),
-    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_release_proxy)
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::__zz_cib_release_proxy),
+    reinterpret_cast<__zz_cib_MTableEntry> (&__zz_cib_Delegator::disruptVTable)
   };
-  static const __zz_cib_MethodTable methodTable = { methodArray, 5 };
+  static const __zz_cib_MethodTable methodTable = { methodArray, 6 };
   return &methodTable;
 }
 }}
 namespace __zz_cib_ { namespace A {
 namespace __zz_cib_Delegator {
