--- 0010-simple-class/src/example-client.cpp
+++ 0030-simple-class-and-forwd-compatibility/src/example-client.cpp
@@ -1,9 +1,20 @@
 #include "example.h"
 
 #include <catch/catch.hpp>
 
+#include <exception>
+
 TEST_CASE("Method call")
 {
   A a;
-  CHECK(a.SomeFunc(10) == 11);
+  if(a.SomeFunc(30) == 32)
+  {
+    // When run with new library
+    CHECK(a.AnotherFunction() == 109);
+  }
+  else
+  {
+    // When run with old library
+    CHECK_THROWS_AS(a.AnotherFunction(), std::bad_function_call);
+  }
 }
