--- pub/example.h
+++ exp/example.h
@@ -1,18 +1,33 @@
 #pragma once
 
+#include "__zz_cib_internal/example-predef.h"
+
 class Interface
 {
 public:
-  virtual int Func() { return 1001; };
-  virtual ~Interface() {}
+  Interface(Interface&& rhs);
+public:
+  Interface();
+  Interface(const Interface&);
+  virtual int Func();
+  virtual ~Interface();
+
+private:
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(Interface, Interface);
 };
 
 class A
 {
 public:
+  A(A&& rhs);
+public:
+  A(const A&);
+  ~A();
   A();
-  int UseInterface(Interface* pInterface) const
-  {
-    return pInterface->Func();
-  }
+  int UseInterface(Interface* pInterface) const;
+
+private:
+  __ZZ_CIB_PROXY_CLASS_INTERNALS(A, A);
 };
+
+#include "__zz_cib_internal/example-postdef.h"
